lesson_05/homework_05.py

```diff
@@ -1,34 +1,71 @@
 small_list = [3, 1, 4, 5, 2, 5, 3]
 big_list = [3, 5, -2, -1, -3, 0, 1, 4, 5, 2]
+
 # task 1. Знайдіть всі унікальні елементи в списку small_list
+unique_elements = set(small_list)
+print("unique ", unique_elements)
 
 # task 2. Знайдіть середнє арифметичне всіх елементів у списку small_list
+average = sum(small_list) / len(small_list)
+print("avg ", average)
 
 # task 3. Перевірте, чи є в списку big_list дублікати
-
+if len(big_list) != len(set(big_list)):
+    print(f"{big_list} has duplicates")
+else:
+    print(f"{big_list} hasn't duplicates")
 
 base_dict = {'contry':'Ukraine', 'continent': 'Europe', 'size': 123}
 add_dict = {"a":1, "b":2, "c":2, "d":3, 'size': 12}
 # task 4. Знайдіть ключ з максимальним значенням у словнику add_dict
+max_key = max(add_dict, key=add_dict.get)
+print("max", max_key)
 
 # task 5. Створіть новий словник, в якому ключі та значення будуть
 # замінені місцями у заданому словнику
+new_dict = {}
+for key, value in add_dict.items():
+    new_dict[value] = key
+print("new dict ", new_dict)
 
-# task 6. Об'єднайте два словника base_dict та add_dict  в новий словник sum_dict
+# task 6. Об'єднайте два словника base_dict та add_dict в новий словник sum_dict
 # Якщо ключі збігаються, то перетворіть значення в строку та об'єднайте їх
 sum_dict = {}
+for key, value in base_dict.items():
+    sum_dict[key] = str(value)
+
+for key, value in add_dict.items():
+    if key in sum_dict:
+        sum_dict[key] += str(value)
+    else:
+        sum_dict[key] = str(value)
+print("sum dict ", sum_dict)
 
 # task 7.
 line = "Створіть множину всіх символів, які входять у заданий рядок"
+character_set = set(line)
+print("set ", character_set)
 
 # task 8. Обчисліть суму елементів двох множин, які не є спільними
 set_1 = {1, 2, 3, 4, 5}
 set_2 = {4, 6, 5, 10}
 
+sum_of_unique_elements = sum(set_1 | set_2)
```

Comment:
Оператор | використовується для об'єднання (злиття) двох множин у нову множину. 
Нам треба знайти відмінні елементи
```suggestion
sum_of_unique_elements = sum(set_1 ^ set_2)
```